pkg/wasm/convert.go

```diff
@@ -179,25 +183,30 @@ func convertWasmConfigFromRemoteToLocal(ec *core.TypedExtensionConfig, wasmHTTPF
 			}
 			pullSecret = []byte(sec)
 		}
-		// Strip all internal env variables from VM env variable.
-		// These env variables are added by Istio control plane and meant to be consumed by the agent for image pulling control,
-		// thus should not be leaked to Envoy or the Wasm extension runtime.
+
+		if ps, found := envs.KeyValues[model.WasmPolicyEnv]; found {
+			if p, found := extensions.PullPolicy_value[ps]; found {
+				pullPolicy = extensions.PullPolicy(p)
+			}
+		}
+		resourceVersion = envs.KeyValues[model.WasmResourceVersionEnv]
+
+		// Strip all internal env variables(with ISTIO_META) from VM env variable.
+		// These env variables are added by Istio control plane and meant to be consumed by the
+		// agent for image pulling control should not be leaked to Envoy or the Wasm extension runtime.
 		delete(envs.KeyValues, model.WasmSecretEnv)
+		for k := range envs.KeyValues {
+			if strings.HasPrefix(k, bootstrap.IstioMetaPrefix) {
+				delete(envs.KeyValues, k)
+			}
+		}
 		if len(envs.KeyValues) == 0 {
 			if len(envs.HostEnvKeys) == 0 {
 				vm.EnvironmentVariables = nil
 			} else {
 				envs.KeyValues = nil
 			}
 		}
-
-		if ps, found := envs.KeyValues[model.WasmPolicyEnv]; found {
```

Comment:
WasmPolicyEnv is at https://github.com/istio/istio/blob/7b24ef916be2375fcb4b599bdd96bb3058b813ac/pilot/pkg/model/extensions.go#L47. Why remove it?

Reply:
Sorry. I did not understand. I am removing all `ISTIO_META` prefixed keys with this change See L198 - so this explicit handling is not needed for this specific key.